includes/modules/order_total/ot_coupon.php

```diff
@@ -148,28 +131,41 @@ function process()
         $this->deduction = $od_amount['total'];
 
         if ($od_amount['total'] > 0) {
+            if ($this->include_shipping === 'true') {
+                $order->info['shipping_cost'] -= $od_amount['shipping'];
+                $order->info['shipping_tax'] -= $od_amount['ShippingTax'];
+            }
             $tax = 0;
+            // Update all tax rates in order object
             foreach ($order->info['tax_groups'] as $key => $value) {
                 if (isset($od_amount['tax_groups'][$key])) {
                     $order->info['tax_groups'][$key] -= $od_amount['tax_groups'][$key];
-                    $order->info['tax_groups'][$key] = zen_round($order->info['tax_groups'][$key], $currencies->get_decimal_places($_SESSION['currency']));
                     $tax += $od_amount['tax_groups'][$key];
+                    $order->info['tax_subtotals'][$key]['subtotal'] -= $od_amount['total'];
```

Comment:
I'm not understanding this.
Why reduce `tax-subtotals` by `$od_amount['total']`, which is the entire discount amount... instead of only by the amount of tax relevant to that specific tax_subtotal?

Reply:
This subtotal is not a tax. Array info['tax_subtotals'] contain the pair 'Tax rate'-'Order subtotal on which tax apply' for each tax class. With these pairs it is possible to recalculate tax value for each tax rate. Array tax_groups is similar and contain tax rate names and values. The second one is used a lot in modules codes, but I am not sure tax_subtotals array is used much.

Reply:
Oh. Yes. Thanks. I knew I was missing something :D